apps/mobile/src/components/account-selector-sheet.tsx

```diff
@@ -0,0 +1,213 @@
+  scrollViewRef: RefObject<ScrollView>;
+  direction: SharedValue<'down' | 'up'>;
+  onAccountPress(accountId: string): void;
+  swapAccountIndexes(idx1: number | null, idx2: number | null): void;
+}) {
+  const ref = useRef<Animated.View>();
+
+  const zoom = useSharedValue(1);
+  const thisItemDimensions = useSharedValue(APPROXIMATE_DEFAULT_ITEM_SIZE);
+  const isAnimatedShared = useSharedValue(false);
+  const translationY = useSharedValue(0);
+  const thisItemPosition = useSharedValue({ x: 0, y: 0 });
+
+  const animatedStyles = useAnimatedStyle(() => ({
+    transform: [{ translateY: translationY.value }, { scale: zoom.value }],
+    position: isAnimatedShared.value ? 'absolute' : 'relative',
+    zIndex: isAnimatedShared.value ? 100 : 1,
+    width: '100%',
+    top: isAnimatedShared.value ? thisItemPosition.value.y : 0,
+    left: isAnimatedShared.value ? thisItemPosition.value.x : 0,
+  }));
+  const placeholderUpAnimatedStyles = useAnimatedStyle(() => ({
+    display: placeholderIdx.value === idx && direction.value === 'up' ? 'flex' : 'none',
+    ...thisItemDimensions.value,
+  }));
+  const placeholderDownAnimatedStyles = useAnimatedStyle(() => ({
+    display: placeholderIdx.value === idx && direction.value === 'down' ? 'flex' : 'none',
+    ...thisItemDimensions.value,
+  }));
+
+  const pan = Gesture.Pan()
+    .activateAfterLongPress(300)
+    .minDistance(1)
+    .onStart(() => {
+      'worklet';
+      zoom.value = withTiming(1.05);
+      isAnimatedShared.value = true;
+      placeholderIdx.value = idx;
+    })
+    .onUpdate(event => {
+      'worklet';
+      translationY.value = event.translationY;
+      const idxDiff =
+        event.translationY >= 0
+          ? Math.floor(event.translationY / thisItemDimensions.value.height)
+          : Math.floor(event.translationY / thisItemDimensions.value.height) + 1;
+
+      direction.value = event.translationY >= 0 ? 'down' : 'up';
+
+      const newIdx = idx + idxDiff;
+      if (newIdx > accountsLength - 1) {
+        placeholderIdx.value = accountsLength - 1;
+      } else if (newIdx < 0) {
+        placeholderIdx.value = 0;
+      } else {
+        placeholderIdx.value = newIdx;
+      }
+    })
+    .onFinalize(() => {
+      zoom.value = 1;
+      isAnimatedShared.value = false;
+      runOnJS(swapAccountIndexes)(idx, placeholderIdx.value);
+
+      placeholderIdx.value = null;
+      translationY.value = 0;
+    });
+
+  const onPress = Gesture.Tap().onEnd(() => {
+    runOnJS(onAccountPress)(accountId);
+  });
+
+  const gestures = Gesture.Exclusive(pan, onPress);
+
+  return (
+    <>
+      <Animated.View style={placeholderUpAnimatedStyles} />
+      <GestureDetector gesture={gestures}>
+        <Animated.View
+          onLayout={e => {
+            const { height, width } = e.nativeEvent.layout;
+            thisItemDimensions.value = { height, width };
+          }}
+          // @ts-expect-error some ref type issues, it should be fine
```

Comment:
😰

Reply:
Can this be explained? I'm not sure we typically do this? 🤔